SeleniumTunnel.js

```diff
@@ -0,0 +1,229 @@
+/**
+ * @module digdug/SeleniumTunnel
+ */
+
+var Tunnel = require('./Tunnel');
+var util = require('./util');
+var pathUtil = require('path');
+var Promise = require('dojo/Promise');
+var descriptors = require('./seleniumDescriptors');
+var fs = require('fs');
+var request = require('dojo/request');
+
+function SeleniumTunnel() {
+	Tunnel.apply(this, arguments);
+
+	if (!this.seleniumDrivers) {
+		this.seleniumDrivers = [ 'chrome' ];
+	}
+}
+
+var _super = Tunnel.prototype;
+SeleniumTunnel.prototype = util.mixin(Object.create(_super), /** @lends module:digdug/SauceLabsTunnel# */ {
+	constructor: SeleniumTunnel,
+
+	/**
+	 * The desired version of selenium to install. This can be defined using a version number or an object containing a
+	 * version number and baseUrl.
+	 *
+	 * example:
+	 * 	{
+	 * 		version: '2.53.0',
+	 * 		baseUrl: 'https://selenium-release.storage.googleapis.com'
+	 * 	}
+	 *
+	 * @type {string|object}
+	 * @default
+	 */
+	seleniumVersion: descriptors.SeleniumConfig.prototype.version,
+
+	/**
+	 * The desired selenium drivers to install. This is a list of driver definitions that may either be a basic string
+	 * or an object.
+	 *
+	 * example:
+	 * 	[
+	 * 		'chrome',
+	 * 		{
+	 * 			name: 'firefox',
+	 * 			version: '0.8.0',
+	 * 			baseUrl: 'https://github.com/mozilla/geckodriver/releases/download'
+	 * 		}
+	 * 	]
+	 *
+	 * @type {Array}
+	 * @default [ 'chrome' ]
+	 */
+	seleniumDrivers: null,
+	
+	startupTimeout: 5000,
+
+	seleniumArgs: null,
+
+	get directory() {
+		return pathUtil.join(__dirname, 'selenium-standalone');
+	},
+
+	get executable() {
+		return 'java';
+	},
+
+	get isDownloaded() {
+		var directory = this.directory;
+		return this._getConfigs().every(function (config) {
+			return fs.existsSync(pathUtil.join(directory, config.executable));
+		});
+	},
+
+	_getDriverConfigs: function () {
+		return this.seleniumDrivers.map(function (driver) {
+			return descriptors.getDriverConfig(driver);
+		});
+	},
+
+	_getConfigs: function () {
+		var configs = this._getDriverConfigs();
+		configs.push(new descriptors.SeleniumConfig(this.seleniumVersion));
+		return configs;
+	},
+
+	download: function (forceDownload) {
+		var tasks = this._getConfigs().map(function (config) {
+			return this._download(config, forceDownload);
+		}.bind(this));
+		
+		return Promise.all(tasks);
+	},
+
+	_download: function (config, forceDownload) {
+		var executable = config.executable;
+		var isDownloaded = fs.existsSync(pathUtil.join(this.directory, executable));
+		var self = {
+			isDownloaded: isDownloaded,
+			url: config.url,
+			proxy: this.proxy,
+			directory: this.directory,
+			executable: executable,
+			dontExtract: !!config.dontExtract
+		};
+		// Leverages the parent download() method by binding a call using the parameters created above
+		// TODO we should consider extracting download into a command/utility function external to Tunnel
```

Comment:
Yeah, I had a thought on that below. I'm not super excited about passing in synthetic `this`.